internal/provider/resource_time_rotating.go

```diff
@@ -0,0 +1,710 @@
+	}
+
+	formattedRotationTimestamp := rotationTimestamp.Format(time.RFC3339)
+
+	return timeRotatingModelV0{
+		RotationRFC3339: types.String{Value: formattedRotationTimestamp},
+		Triggers:        plan.Triggers,
+		Year:            types.Int64{Value: int64(rotationTimestamp.Year())},
+		Month:           types.Int64{Value: int64(rotationTimestamp.Month())},
+		Day:             types.Int64{Value: int64(rotationTimestamp.Day())},
+		Hour:            types.Int64{Value: int64(rotationTimestamp.Hour())},
+		Minute:          types.Int64{Value: int64(rotationTimestamp.Minute())},
+		Second:          types.Int64{Value: int64(rotationTimestamp.Second())},
+		RotationYears:   plan.RotationYears,
+		RotationMonths:  plan.RotationMonths,
+		RotationDays:    plan.RotationDays,
+		RotationHours:   plan.RotationHours,
+		RotationMinutes: plan.RotationMinutes,
+		RFC3339:         types.String{Value: formattedTimestamp},
+		Unix:            types.Int64{Value: rotationTimestamp.Unix()},
+		ID:              types.String{Value: formattedTimestamp},
+	}, nil
+
+}
+
+func parseTwoPartId(idParts []string, resp *resource.ImportStateResponse) timeRotatingModelV0 {
+
+	baseRfc3339 := idParts[0]
+	rotationRfc3339 := idParts[1]
+
+	timestamp, err := time.Parse(time.RFC3339, baseRfc3339)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Import Timestamp Error",
+			"The baseRfc3339 timestamp that was supplied could not be parsed as RFC3339.\n\n+"+
+				fmt.Sprintf("Original Error: %s", err),
+		)
+		return timeRotatingModelV0{}
+	}
+
+	rotationTimestamp, err := time.Parse(time.RFC3339, rotationRfc3339)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Import Timestamp Error",
+			"The rotationRfc3339 timestamp that was supplied could not be parsed as RFC3339.\n\n+"+
+				fmt.Sprintf("Original Error: %s", err),
+		)
+		return timeRotatingModelV0{}
+	}
+
+	formattedTimestamp := timestamp.Format(time.RFC3339)
+
+	return timeRotatingModelV0{
+		Year:            types.Int64{Value: int64(rotationTimestamp.Year())},
+		Month:           types.Int64{Value: int64(rotationTimestamp.Month())},
+		Day:             types.Int64{Value: int64(rotationTimestamp.Day())},
+		Hour:            types.Int64{Value: int64(rotationTimestamp.Hour())},
+		Minute:          types.Int64{Value: int64(rotationTimestamp.Minute())},
+		Second:          types.Int64{Value: int64(rotationTimestamp.Second())},
+		RotationRFC3339: types.String{Value: rotationRfc3339},
+		RotationYears:   types.Int64{Null: true},
+		RotationMonths:  types.Int64{Null: true},
+		RotationDays:    types.Int64{Null: true},
+		RotationHours:   types.Int64{Null: true},
+		RotationMinutes: types.Int64{Null: true},
+		RFC3339:         types.String{Value: formattedTimestamp},
+		Unix:            types.Int64{Value: rotationTimestamp.Unix()},
+		ID:              types.String{Value: formattedTimestamp},
+	}
+}
+
+func parseMultiplePartId(idParts []string, resp *resource.ImportStateResponse) timeRotatingModelV0 {
```

Comment:
Consider removing `resp *resource.ImportStateResponse` as argument and returning `(timeRotatingModelV0, error)`.